### ISSUE TITLE
Make vertical build defines of Host and Target properties explicit

### ISSUE THREAD

**Issue by mmitche** [dotnet/source-build#4143]
When building in CI/Official and PRs, we should make our Host and Target properties explicit on the command line:

- HostArchitecture
- TargetArchitecture
- HostOS
- TargetOS
- HostRid
- TargetRid

Much of the time, many of these properties will match. But there is quite a lot of implicit behavior when inferring them in dev scenarios.

**Comment by mmitche** [comment]
See https://github.com/dotnet/installer/pull/18658 for extended conversation on this topic.

**Comment by tmds** [comment]
note: `TargetOS` as how the runtime repo uses it is not the 'os' part of the `TargetRid`. For non-portable Linux builds, `linux-bionic`, and `linux-musl`, `TargetOS` is `linux`. And on Windows, it is `windows` instead of `win`.

**Comment by tmds** [comment]
> TargetOS as how the runtime repo uses it is not the 'os' part of the TargetRid.

The same is true for TargetOS in the vmr.

Having a property that contains the 'os' part of the `TargetRid` wouldn't be very useful to have. `TargetRid` is _just the name_ for what gets built.

`TargetOS` tells us something meaningful about the target: _the target is Linux_.

**Comment by ViktorHofer** [comment]
> as how the runtime repo uses it is not the 'os' part of the TargetRid. For non-portable Linux builds, linux-bionic, and linux-musl, TargetOS is linux

Right, @directhex mentioned that in the past. Ideally TargetOS would be the full OS string, i.e. `linux-musl` or `linux-bionic`. We will always have one special case for `windows` vs `win` and I think that's fine. It would be good to eventually clean this up in runtime and also make the runtime build use the correct properties.

**Comment by ViktorHofer** [comment]
This is now already implemented for the Target* properties. We didn't see a need to also flow the Host properties in so far. The repos that need them currently calculate them individually. I think we can close this.
